### PR TITLE
Allow source ID ot be saved as FHIR logical ID, map between source id and interaction id

### PR DESCRIPTION
**Pull request by rcrichton**
This pull request adds functionality to save by source ID and swaps the source ID for the interaction ID in the FHIR Patient endpoint.

To use this OpenHIM channels must also be edited. The JeMPI channel patttern must be changed to `/fhir/Patient\??[^/]*` to allow the MPI mediator to handle `/Patient/<surceId>` queries.

This change has also been push to a temporary tag on docker hub for testing: `sl-conn`

### CODE REVIEW

**Review by BMartinos — COMMENTED** [review]
Overall the changes looks good to me :+1: 
I have been able to successfully look up the patient on /fhir/Patient/UUID (supplied in the creation). 
The supplied UUID is able to be used on the various other FHIR requests.

Still just doing a couple of other test cases which ive run into some issues (mostly edge cases), but all happy paths requests are looking good

File: src/routes/index.ts [hunk 1]
```diff
@@ -61,6 +64,75 @@ routes.post(
...
+  } = getConfig();
+  const fhirResponse = await getData(
+    fhirProtocol,
+    fhirHost,
+    fhirPort,
+    `/fhir/Patient/${requestedId}`,
+    {}
+  );
+
+  let upsteamId = requestedId;
```
Comment (MatthewErispe) [inline src/routes/index.ts:90]: @bradsawadye typo here, should be upstreamId?

**Review by MatthewErispe — CHANGES_REQUESTED** [review]
@bradsawadye ltgm apart from one typo

**Review by MatthewErispe — APPROVED** [review]

**Review by brett-onions — COMMENTED** [review]
LGTM
